refactor(docs) Extend table comparing `Simulation` vs `Deployment` runtimes (#5375)
adap/flower: framework/docs/source/ref-simulation-deployment-runtimes.rst

Simulation vs. Deployment Runtime
=================================

Flower Apps can be executed with both with `simulation <how-to-run-simulations.html>`_
or `deployment <how-to-run-flower-with-deployment-engine.html>`_ runtimes. Switching
between runtimes simply requires specifying a different type of `federation` when
executing the `flwr run` command via the `Flower CLI <ref-api-cli.html>`_.

While Flower Apps can run in both simulation and deployment without making changes to
their code, there are some differences on how they get executed depending on the

> runtime. The following table outlines the key characteristics that differentiate
> simulated Flower federations from deployed ones.

Author: ```suggestion
runtime. The following table outlines the key characteristics that differ when
executing a FlowerApp in a Flower federation with the simulation runtime from another using the deployment runtime.
```

Author: @dstripelis I think like this is clearer. Technically speaking federations aren't executed.

> simulated Flower federations from deployed ones.

Reviewer: Adding this comment in the latest commit.

> simulated Flower federations from deployed ones.

Reviewer: Incorporating the suggestion below: https://github.com/adap/flower/pull/5375#discussion_r2099837969

      - Deployment Runtime
    - - **Lifecycle Stage**
      - Ideal for rapid prototyping, algorithm validation, research, debugging, and
        experimentation.
      - Deploy validated use cases in production, real-world privacy-preserving
        applications.
    - - **Environment**
      - Local or remote, standalone, controlled.
      - Distributed, remote.
    - - **Data**
      - Simulated data partitions, public or private datasets or artificially generated
        - a natural fit for `Flower Datasets <https://flower.ai/docs/datasets/>`_.
      - Real client-side data, residing on local databases or filesystems.
    - - **Backend**
      - Multiple Python processes/workers coordinated using `Ray
        <https://docs.ray.io/>`_.
      - Multiple independent processes or subprocesses running in coordination with the
        SuperLink and SuperNodes.
    - - **Execution Mode**

>       - Multiprocessing execution where each process simulates being a distinct client.

Author: ```suggestion
      - Multiprocessing execution where each process simulates a distinct client.
```

      - Parallel execution mode across a network of physical machines/devices or
        computing environment.
    - - **Communication**

>       - Managed via in-memory communication.
>       - Managed via Flower Deployment Runtime through TLS-enabled gRPC.

Author: ```suggestion
      - In-memory communication.
      - TLS-enabled gRPC.
```

    - - **Server-side Infrastructure**
      - Simulation runtime coordinates the spawning of workers (python process) which
        play the role of `simulated` SuperNodes. There is no SuperLink.
      - The SuperLink awaits for SuperNodes to connect. User interface with the
        SuperLink using the `Flower CLI <ref-api-cli.html>`_.
    - - **Server-side App execution**
      - A ``ServerApp`` process is initialized inside a controlled environment and
        communicates in-memory with workers.
      - ``ServerApp`` `process or subprocess <ref-flower-network-communication.html>`_
        runs independently from the SuperLink and communicates with it over gRPC via the
        ServerAppIO API.
    - - **Client-side Infrastructure**

>       - None. The Simulation runtime is standalone.

Author: ```suggestion
      - None. The simulation runtime is self-contained.
```
